service/matching/task_queue_partition_manager.go

```diff
@@ -93,6 +131,30 @@ func newTaskQueuePartitionManager(
 	return pm, nil
 }
 
+// UpdateRatelimit updates the task dispatch rate
+func (pm *taskQueuePartitionManagerImpl) UpdateRatelimit(rps float64) {
+	nPartitions := float64(pm.config.NumReadPartitions())
+	if nPartitions > 0 {
+		// divide the rate equally across all partitions
+		rps = rps / nPartitions
+	}
+	burst := int(math.Ceil(rps))
+
+	minTaskThrottlingBurstSize := pm.config.MinTaskThrottlingBurstSize()
+	if burst < minTaskThrottlingBurstSize {
+		burst = minTaskThrottlingBurstSize
+	}
+
+	pm.dynamicRateBurst.SetRPS(rps)
+	pm.dynamicRateBurst.SetBurst(burst)
```

Comment:
it looks like we can do
```suggestion
	pm.dynamicRateBurst.SetRateBurst(rps, burst)
```
to be more efficient (only lock once)

Reply:
looks like `SetRateBurst` is not part of the `dynamicRateBurst` interface

Reply:
Nevermind, I see that for MutableRateBurstImpl it doesn't matter